main/src/main/java/org/mobilitydata/gtfsvalidator/validator/NetworkIdConsistencyValidator.java

```diff
@@ -0,0 +1,175 @@
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static org.mobilitydata.gtfsvalidator.notice.SeverityLevel.ERROR;
+
+import java.util.*;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidationNotice;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.*;
+
+@GtfsValidator
+public class NetworkIdConsistencyValidator extends FileValidator {
+  private final GtfsRouteTableContainer routeTableContainer;
+  private final GtfsRouteNetworkTableContainer routeNetworkTableContainer;
+
+  private final GtfsNetworkTableContainer networkTableContainer;
+
+  @Inject
+  NetworkIdConsistencyValidator(
+      GtfsRouteTableContainer routes,
+      GtfsRouteNetworkTableContainer routeNetworks,
+      GtfsNetworkTableContainer networks) {
+    this.routeTableContainer = routes;
+    this.routeNetworkTableContainer = routeNetworks;
+    this.networkTableContainer = networks;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    boolean hasNetworkIdField = this.routeTableContainer.hasColumn(GtfsRoute.NETWORK_ID_FIELD_NAME);
+
+    if (hasNetworkIdField && !this.routeNetworkTableContainer.isMissingFile()) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsRouteNetwork.FILENAME));
+    }
+    if (!this.networkTableContainer.isMissingFile() && hasNetworkIdField) {
+      noticeContainer.addValidationNotice(
+          new ConditionalForbiddenFileNotice(
+              GtfsRoute.FILENAME, GtfsRoute.NETWORK_ID_FIELD_NAME, GtfsNetwork.FILENAME));
+    }
```

Comment:
This is a subtle point, but should route_networks.txt and networks.txt be conditionally forbidden if routes.txt:network_id is specified?  Or should it be vice versa?  The spec claims both directions and I'd selfishly rather people use route_networks.txt, but I'm not sure the validator should be stating a preference?

Which is to say, instead of a generic `ConditionallyForbiddenFileNotice`, maybe we should have a specific `RouteNetworksSpecifiedInMoreThanOneFile` or something similar that can give more details on the situation?

Reply:
✅ Done
@emmambd I had to change the notice details to match this new notice definition. Could you validate?